src/index.js

```diff
@@ -1,20 +1,117 @@
-function server(appConfig, escapeHatch = null) {
-    const baseConfig = require('./config.js');
-    const config = Object.assign({}, baseConfig, appConfig);
-    const log = global.log = require('./logging.js')(config.logging);
+const http = require('http');
+const express = require('express');
+const reqId = require('express-request-id')();
+const logger = require('cnn-logger');
+const defaultConfig = require('./config');
 
-    require('./check-node-environment.js')();
+function reqLogger(req, res, next) {
+  req.log.info(`Request: ${req.id} ${req.method} ${req.originalUrl}`);
+  res.on('finish', function() {
+    req.profile.end = new Date();
+    req.profile.duration = req.profile.end - req.profile.start;
+    req.log.info(`Request: ${req.id} completed. Duration: ${req.profile.duration} ms`);
+  });
+  next();
+}
+
+class Server {
+  constructor(config) {
+    this.config = config || defaultConfig;
+    this.log = logger(this.config.logging);
+    this.app = express();
+    this.app.use((req, res, next) => {
+      req.profile = { start: new Date() };
+      req.log = this.log;
+      if (this.config.messenger && this.config.messenger.amqp) {
+         req.messenger = this.messenger;
+         req.Message = this.Message;
+      }
+      next();
+    });
+    this.app.disable('x-powered-by');
+    this.app.use(reqId);
+    this.app.use(reqLogger);
+    this.service = null;
+  }
+
+  async start() {
```

Comment:
node.js doesn't support async await fully until 7.10.1 [http://node.green/](node.green). can we put in a .nvmrc file with the supported node version?